modules/ting_search/modules/subsearch_kpi/subsearch_kpi.module

```diff
@@ -0,0 +1,220 @@
+  $password = variable_get('subsearch_kpi_password', NULL);
+
+  if (empty($url) || empty($account) || empty($password)) {
+    watchdog('subsearch_kpi', 'Module not configured properly', array(), WATCHDOG_WARNING);
+    return;
+  }
+
+  try {
+    $client = new HttpClient();
+    $response = $client->get(
+      $url,
+      array(
+        'query' => $params,
+        'headers' => array(
+          'Accept' => 'application/json'
+        ),
+        'auth' => [
+          $account,
+          $password
+        ]
+      )
+    );
+    $result = json_decode($response->getBody());
+  } catch (GuzzleClientException $e) {
+    throw new Exception($e->getMessage(), $e->getCode(), $e);
+  }
+
+  if (!empty($result)) {
+    subsearch_kpi_insert_records($result);
+    variable_set('subsearch_kpi_cron_run', time() + (60*60*24));
+    watchdog('subsearch_kpi', 'Data fetched from KPI. Will re-run in day.', array(), WATCHDOG_INFO);
+  }
+  else {
+    watchdog('subsearch_kpi', 'No data fetched from feed', array(), WATCHDOG_WARNING);
+  }
+}
+
+/**
+ * Insert kpi keys into table.
+ */
+function subsearch_kpi_insert_records($json) {
+
+  db_truncate('kpi_index_popular_keys')->execute();
+
+  foreach ($json->data[0]->SubRows[0] as $keys => $data) {
+    $query = db_insert('kpi_index_popular_keys')->fields(
+      array(
+        'id' => md5($keys),
+        'popular_keys' => $keys,
+        'numtotalsearches' => (int) $data->measures->Searches,
+      )
+    );
+    $query->execute();
+  }
+}
+
+/**
+ * Implements hook_libraries_info().
+ */
+function subsearch_kpi_libraries_info() {
+  return array(
+    'guzzle' => array(
+      'name' => 'Guzzle',
+      'vendor url' => 'https://github.com/guzzle/guzzle',
+      'download url' => 'https://github.com/guzzle/guzzle',
+      'version' => '6.2',
+      'xautoload' => function ($adapter) {
+        $adapter->composerJson('composer.json');
+      },
+    ),
+    'promises' => array(
+      'name' => 'Guzzle promises library',
+      'vendor url' => 'https://github.com/guzzle/promises',
+      'download url' => 'https://github.com/guzzle/promises',
+      'version' => '1.2',
+      'xautoload' => function ($adapter) {
+        $adapter->composerJson('composer.json');
+      },
+    ),
+    'http-message' => array(
+      'name' => 'Common interface for HTTP messages',
+      'vendor url' => 'https://github.com/php-fig/http-message',
+      'download url' => 'https://github.com/php-fig/http-message',
+      'version' => '1.0',
+      'xautoload' => function ($adapter) {
+        $adapter->composerJson('composer.json');
+      },
+    ),
+    'psr7' => array(
+      'name' => 'PSR-7 message implementation',
+      'vendor url' => 'https://github.com/guzzle/psr7',
+      'download url' => 'https://github.com/guzzle/psr7',
+      'version' => '1.3',
+      'xautoload' => function ($adapter) {
+        $adapter->composerJson('composer.json');
+      },
+    ),
```

Comment:
Are all these libraries necessary to declare for GuzzleHttp to function properly?